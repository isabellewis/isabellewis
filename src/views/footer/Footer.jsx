import React from "react";
import styled from "styled-components";

import { SocialMedia } from "../../components";
import { ShortCopy } from "../../elements";

const FooterPanel = styled.div`
  background: #fff;
  bottom: 0;
  left: 0;
  padding: 10px;
  position: fixed;
  width: 100%;
`;

const CopyStyling = styled.div`
  color: #bbb;
  text-align: center;
`;

export const Footer = () => (
  <FooterPanel>
    <CopyStyling>
      <ShortCopy theme="light">Fancy a chat?</ShortCopy>
    </CopyStyling>
    <SocialMedia theme="footer" />
  </FooterPanel>
);
