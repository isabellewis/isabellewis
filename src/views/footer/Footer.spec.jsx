import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Footer } from './Footer';

describe('views/Footer', () => {
	let component;

	beforeEach(() => {
		component = shallow(<Footer />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});