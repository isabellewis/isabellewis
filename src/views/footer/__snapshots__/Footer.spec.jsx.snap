// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Footer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <ShortCopy
            shortCopyText="Fancy a chat?"
            theme="light"
          />
        </ForwardRef>,
        <SocialMedia
          theme="footer"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ShortCopy
            shortCopyText="Fancy a chat?"
            theme="light"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "shortCopyText": "Fancy a chat?",
            "theme": "light",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Footer__CopyStyling-bubGHg",
            "isStatic": true,
            "rules": Array [
              "color:#bbb;text-align:center;",
            ],
          },
          "displayName": "Footer__CopyStyling",
          "render": [Function],
          "styledComponentId": "Footer__CopyStyling-bubGHg",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "theme": "footer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Footer__FooterPanel-ksWeIc",
        "isStatic": true,
        "rules": Array [
          "background:#fff;padding:10px;width:100%;",
        ],
      },
      "displayName": "Footer__FooterPanel",
      "render": [Function],
      "styledComponentId": "Footer__FooterPanel-ksWeIc",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <ShortCopy
              shortCopyText="Fancy a chat?"
              theme="light"
            />
          </ForwardRef>,
          <SocialMedia
            theme="footer"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ShortCopy
              shortCopyText="Fancy a chat?"
              theme="light"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "shortCopyText": "Fancy a chat?",
              "theme": "light",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Footer__CopyStyling-bubGHg",
              "isStatic": true,
              "rules": Array [
                "color:#bbb;text-align:center;",
              ],
            },
            "displayName": "Footer__CopyStyling",
            "render": [Function],
            "styledComponentId": "Footer__CopyStyling-bubGHg",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "theme": "footer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "Footer__FooterPanel-ksWeIc",
          "isStatic": true,
          "rules": Array [
            "background:#fff;padding:10px;width:100%;",
          ],
        },
        "displayName": "Footer__FooterPanel",
        "render": [Function],
        "styledComponentId": "Footer__FooterPanel-ksWeIc",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
